gender_analysis/common.py

```diff
@@ -111,124 +112,50 @@
-        if isinstance(file_path, str):
-            file_path = Path(file_path)
-
-        # make sure that we only try to load supported file types
-        supported_file_types = {'.csv', '.txt'}
-        current_file_type = file_path.parts[-1][file_path.parts[-1].rfind('.'):]
-        if current_file_type not in supported_file_types:
-            err = "The FileLoaderMixin currently supports "
-            err += "{supported_file_types} but not {current_file_type}."
-            raise ValueError(err)
-
-        # check if we are working locally and in the correct directory
-        # __file__ is only available if executed from a file but
-        # not from an ipython shell or notebook
-        # in those cases, the file has to be loaded remotely from github.
-        try:
-            local_path = os.path.abspath(os.path.dirname(__file__))
-            is_local = True
-            if not local_path.endswith('gender_analysis'):
-                is_local = False
-                warning = "WARNING: The FileLoaderMixin should be placed "
-                warning += "in the main path of the gender_analysis project."
-                warning += f"It's currently in {local_path}. Until the Mixin "
-                warning += "is in the correct path, files are loaded "
-                warning += "from Github."
-                print(warning)
-        except NameError:
-            is_local = False
-
-        if is_local:
-            if DEBUG:
-                print(f'loading {file_path} locally.')
-            return self.load_file_locally(file_path, current_file_type)
-        else:
-            if DEBUG:
-                print(f'loading {file_path} remotely')
-            return self.load_file_remotely(file_path, current_file_type)
-
-    @staticmethod
-    def load_file_locally(file_path, current_file_type):
-        # I need a way of getting the local path to the base of the repo.
-        # This file is currently in the base of the
-        # repo so it returns the correct path. But it will change once
-        # this function gets moved.
-        local_base_path = Path(os.path.abspath(os.path.dirname(__file__)))
-        file = open(local_base_path.joinpath(file_path), encoding='utf-8')
-
-        if current_file_type == '.csv':
-            result = file.readlines()
-        elif current_file_type == '.txt':
-            try:
-                result = file.read()
-            except UnicodeDecodeError as err:
-                print(f'File loading error with {file_path}.')
-                raise err
-
-        else:
-            raise Exception(
-                'Cannot load if current_file_type is not .csv or .txt')
-
-        file.close()
-        return result
+    :param file_path: can be a string or Path object
+    :return: the text as a string type
+    """
+    if isinstance(file_path, str):
+        file_path = Path(file_path)
 
+    file_type = file_path.parts[-1][file_path.parts[-1].rfind('.'):]
```

Comment:
All of the same comments about load_csv_to_dict apply here, except that the function name is fine!